src/components/viewmodels/settings/encryption/KeyStoragePanelViewModel.ts

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2025 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only OR LicenseRef-Element-Commercial
+Please see LICENSE files in the repository root for full details.
+*/
+
+import { useCallback, useEffect, useState } from "react";
+import { logger } from "matrix-js-sdk/src/logger";
+
+import { useMatrixClientContext } from "../../../../contexts/MatrixClientContext";
+
+interface KeyStoragePanelState {
+    /**
+     * Whether the app's "key storage" option should show as enabled to the user,
+     * or 'undefined' if the state is still loading.
+     */
```

Comment:
https://github.com/element-hq/element-web/pull/29113/files#r1956081855 asked to make this clearer: I've added details but I'd argue that if this is asking for exactly what state it means the keys are in, that detail doesn't belong here: it's a view controller, so its job is to specify how the UI looks, not what that might happen to imply under the hood.